client/login/magic-login/index.jsx

```diff
@@ -146,6 +148,8 @@ class MagicLogin extends Component {
 			} );
 		}
 
+		this.isInitialMount = false;
```

Comment:
This looks like a code smell. Is there really no better way to determine whether to show the loading ellipsis?

Reply:
I agree that this solution doesn't seem ideal. I couldn't find any state variable or loading state or prop that would prevent at least a small flash of the form before the email send was requested. All of the loading statuses are false by default for a moment and it would cause a very quick flash of it before the Loading Ellipses showed instead.

There may be a better way if there's a new state variable loaded or if default loading states are changed but I know this class is used for many different flows and didn't want to risk affecting any of them unnecessarily. That and because it was quite a rush to get this flow ready I figured this was the best way for now. I can add to the project for i2 to potentially look at a more comprehensive fix for this. Does that sound okay?

Reply:
@tyxla I had a bit of unexpected free time today so I whipped up another solution [here](https://github.com/Automattic/wp-calypso/pull/102246)

Essentially it is still a check to make sure the form is mounted, but done in the state object rather than a class variable outside of the state. I still think it would be more ideal to have a specific, existing, loading state to check to show the loading ellipses but I am unsure that exists in its current state. I think the PR is minor improvement, let me know what you think 😄

Reply:
Thanks!